state/execution.go

```diff
@@ -152,11 +152,12 @@ func (blockExec *BlockExecutor) CreateProposalBlock(
 
 	// update the block with the response from PrepareProposal
 	block.Data, _ = types.DataFromProto(&cmtproto.Data{
-		SquareSize: binary.BigEndian.Uint64(rpp.Txs[len(rpp.Txs)-1]),
-		Txs:        rpp.Txs[:len(rpp.Txs)-2],
-		Hash:       rpp.Txs[len(rpp.Txs)-2],
+		Txs: rpp.Txs[:len(rpp.Txs)-2],
```

Comment:
There is no need for the application to pass back the square size to consensus as now it has no more need of it. We can thus modify this so that only the last value is the data hash and the rest are considered the transactions